### ISSUE TITLE
[java] CloneMethodMustBePublic gives false positives

### ISSUE THREAD

**Issue by Noah0120** [pmd/pmd#3440]
**Affects PMD Version:** 
6.36.0
**Rule:**
CloneMethodMustBePublic
Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/latest/pmd_rules_java_errorprone.html#clonemethodmustbepublic

**Description:**
Java manual says that
```
By convention, classes that implement this interface should override 
Object.clone (which is protected) with a public method. 
See Object.clone() for details on overriding this method.
``` 
However, it detects the `protected clone()` method when overriding the method.

**Code Sample demonstrating the issue:**

```
@Override
    protected Object clone() throws CloneNotSupportedException {
        // TODO Auto-generated method stub
        return super.clone();
    }

```

**Expected outcome:**
It should not give any detection results because `clone()` method is `protected`.

**Running PMD through:** *[CLI]*

**Comment by oowekyala** [comment]
Hi, I think there is a misunderstanding. As the name of the rule suggests, an override of the clone method must be declared public (even though the method in Object is declared protected). This is also what the quote you provided says.
